calico/felix/fetcd.py

```diff
@@ -334,6 +335,56 @@ def __init__(self, config, etcd_api, status_reporter, hosts_ipset):
         self.last_rate_log_time = monotonic_time()
         # Register for events when values change.
         self._register_paths()
+        self._usage_report_greenlet = gevent.Greenlet(
+            self._periodically_usage_report
+        )
+
+    def estimated_host_count(self):
+        """
+        Single operations to python's built-in datastructures are thread-safe.
+        You can update a map, or add something to a set or update an integer,
+        and it's atomic.
+
+        :return An aproximation to our Felix cluster count
+        """
+        return len(self.ipv4_by_hostname)
+
+    @logging_exceptions
+    def _periodically_usage_report(self):
+        """
+        Greenlet: periodically report the cluster existence to
+        projectcalico.org.  Period is about once per day.
+
+        :return: Does not return, unless USAGE_REPORT disabled.
+        """
+        interval = 86400   # Once every 24 hours minus 12 minute jitter
+        if not self._config.USAGE_REPORT:
+            _log.info("usage report is disabled")
+            return
+
+        _log.info("Started usage report thread.  Usage report interval: %s", interval)
+
+        while True:
+            try:
+                config = self._config
+                calico_version = str(pkg_resources.require("calico")[0].version)
+                felix_count = self.estimated_host_count()
+                cluster_type = "NA"
+                report_usage_and_get_warnings(calico_version, config.HOSTNAME, config.CLUSTER_GUID, felix_count, cluster_type)
+            except Exception:
+                # except Exception: so that we don't catch SystemExit an KeyboardInterrupt. Those should kill the app.
+
+                # Jitter by 1% of interval (about 12 minutes)
+                jitter = random.random() * 0.01 * interval
+                sleep_time = interval - jitter
+                _log.warning("Error when usage_report. "
```

Comment:
That's a fair point.  Reduced this to info level.  It will not be verbose, since it will only happen once every 24 hours.